### ISSUE TITLE
Placeholder update during git checkout takes a long time

### ISSUE THREAD

**Issue by turbonaitis** [microsoft/VFSForGit#828]
```
[2019-02-20 21:29:15 +02:00] ClearNegativePathCache_ClearedCache {"Area":"GitIndexProjection","Message":"ClearNegativePathCache: Cleared negative path cache","totalEntryCount":0,"negativePathCacheUpdatedForGitCount":0,"clearCacheResult.Result":"GVFS.Virtualization.FileSystem.FileSystemResult","clearCacheResult.RawResult":1}
[2019-02-20 21:29:22 +02:00] UpdatePlaceholders (Start) {"File placeholder count":117135,"Folder placeholders count":29978}
[2019-02-20 21:29:25 +02:00] Warning {"WarningMessage":"WaitToAcquireGVFSLock: BackgroundFileSystemTaskRunner unable to acquire lock, retrying"}
[2019-02-20 21:29:26 +02:00] UpdatePlaceholders (Stop) {"DurationMs":4103}
```
Loading PlaceholderListDatabase took 7s, updating placeholders took another 4s. This is just one sample, but this behavior is reproducable with this amount of hydrated files.

**Comment by turbonaitis** [comment]
@jrbriggs are you guys seeing any such behavior on your side, when the number of placeholders is large? Or is that not a concern for your use cases?

**Comment by jrbriggs** [comment]
We haven't seen this perf problem yet @turbonaitis, but we have some suspicions about what might cause this, and it will likely apply to our large Mac repos too.

**Comment by jrbriggs** [comment]
Related: #873

**Comment by turbonaitis** [comment]
Not sure - #873 mostly talks about the placeholder files themselves, if I'm reading it correctly, while in this case the bottleneck seems to be more related to the implementation of `PlaceholderListDatabase`. I'm working on a PR to address this, hope to submit in the next couple of days (but essentially I'm switching from text format to binary format for `PlaceholderListDatabase`). We can discuss the solution, once the code is there.

**Comment by turbonaitis** [comment]
Also, it would seem to me, that performance should be just as bad on Windows, as the code is shared.

**Comment by jrbriggs** [comment]
Closed with #1140
